monads/src/HW/Eval.hs

```diff
@@ -29,8 +29,44 @@ initialState = MachineState [] M.empty
 -- Execute a single instruction. 
 -- Successful evaluation does not produce any useful result: only the effect of modifying state matters. 
 execInstr :: (Ord v, Show v) => StackInstr v -> MyState (MachineState v) (Either (Error v) ())
-execInstr = undefined 
+execInstr (PushNum x) = do 
+  state <- get 
+  put (state {getStack = x : getStack state, getEnv = getEnv state})
+  return (Right ())
```

Comment:
It's better not to get the whole state, when you only modify a part of it. Try using `modify` instead. 
Furthermore, you can avoid `return` by fmapping `Right`